api/modelmanager/modelmanager.go

```diff
@@ -292,6 +292,14 @@ func (c *Client) ModelInfo(tags []names.ModelTag) ([]params.ModelInfoResult, err
 	if len(results.Results) != len(tags) {
 		return nil, errors.Errorf("expected %d result(s), got %d", len(tags), len(results.Results))
 	}
+	for i := range results.Results {
+		if results.Results[i].Error != nil {
+			continue
+		}
+		if results.Results[i].Result.Type == "" {
+			results.Results[i].Result.Type = "iaas"
```

Comment:
Can this be using a constant rather than direct string literal?

Reply:
Sadly we tend to use literals everywhere :-(
I'll see if I can find a place to define something.

Reply:
hmm, try to refactor juju/juju/state/model.go? It has model types defined. Like ModelTypeIAAS, etc...

Reply:
Also, since you are fixing this in api and not in apiserver layer, please add a comment as to why. It might be helpful to have a bug reference in the comment too.